packages/cli/src/modules/data-store/data-store-column.repository.ts

```diff
@@ -0,0 +1,141 @@
+		return await this.manager.transaction(async (em) => {
+			const existingColumnMatch = await em.existsBy(DataStoreColumnEntity, {
+				name: schema.name,
+				dataStoreId,
+			});
+
+			if (existingColumnMatch) {
+				throw new UserError(
+					`column name '${schema.name}' already taken in data store '${dataStoreId}'`,
+				);
+			}
+
+			if (schema.columnIndex === undefined) {
+				const columns = await this.getColumns(dataStoreId, em);
+				schema.columnIndex = columns.length;
+			} else {
+				await this.shiftColumns(dataStoreId, schema.columnIndex, 1, em);
+			}
+
+			const column = em.create(DataStoreColumnEntity, {
+				...schema,
+				dataStoreId,
+			});
+
+			await em.insert(DataStoreColumnEntity, column);
+
+			// Create user table if it does not exist
+			const tableName = toTableName(dataStoreId);
+			const queryRunner = em.queryRunner;
+			if (!queryRunner) {
+				throw new Error('QueryRunner is not available');
+			}
+			const tableExists = await queryRunner.hasTable(tableName);
+			if (!tableExists) {
+				const dslColumns = [
+					new DslColumn('id').int.autoGenerate2.primary,
+					...toDslColumns([column]),
+				];
+				const createTable = new CreateTable(tableName, '', queryRunner);
+				createTable.withColumns.apply(createTable, dslColumns);
+				await createTable.execute(queryRunner);
+			} else {
+				await em.query(addColumnQuery(tableName, column, em.connection.options.type));
+			}
+
+			return column;
+		});
+	}
+
+	async deleteColumn(dataStoreId: string, column: DataStoreColumnEntity) {
+		await this.manager.transaction(async (em) => {
+			await em.remove(DataStoreColumnEntity, column);
+			await em.query(
+				deleteColumnQuery(toTableName(dataStoreId), column.name, em.connection.options.type),
+			);
+			await this.shiftColumns(dataStoreId, column.columnIndex, -1, em);
+		});
+	}
+
+	async moveColumn(rawDataStoreId: string, columnId: string, targetIndex: number) {
+		await this.manager.transaction(async (em) => {
+			const columnCount = await em.countBy(DataStoreColumnEntity, { dataStoreId: rawDataStoreId });
+
+			if (targetIndex >= columnCount) {
+				throw new UserError('tried to move column to index larger than column count');
+			}
+
+			if (targetIndex < 0) {
+				throw new UserError('tried to move column to negative index');
+			}
+
+			const existingColumn = await em.findOneBy(DataStoreColumnEntity, {
+				id: columnId,
+				dataStoreId: rawDataStoreId,
+			});
+
+			if (existingColumn === null) {
+				throw new UserError(`tried to move column not present in data store '${rawDataStoreId}'`);
+			}
+
+			await this.shiftColumns(rawDataStoreId, existingColumn.columnIndex, -1, em);
+			await this.shiftColumns(rawDataStoreId, targetIndex, 1, em);
+			await em.update(
+				DataStoreColumnEntity,
+				{ id: existingColumn.id },
+				{ columnIndex: targetIndex },
+			);
+		});
+	}
+
+	async shiftColumns(
+		rawDataStoreId: string,
+		lowestIndex: number,
+		delta: -1 | 1,
+		em?: EntityManager,
```

Comment:
seems like we're always passing `this.manager` for this so we can drop the param?